### ISSUE TITLE
Issue with extern "C" inlines being forced multiline

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1259]
things like

```extern "C" NSBundle*            UnityGetMetalBundle()       {
    return _MetalBundle;
}
<...>
extern "C" EAGLContext*         UnityGetDataContextEAGL()   {
    return _GlesContext;
}
```

if i try to change it to be single line like all other functions here it reverts back

```diff
-extern "C" NSBundle*            UnityGetMetalBundle()       { return _MetalBundle; }
+extern "C" NSBundle*            UnityGetMetalBundle()       {
+    return _MetalBundle;
+}
 extern "C" MTLDeviceRef         UnityGetMetalDevice()       { return _MetalDevice; }
 extern "C" MTLCommandQueueRef   UnityGetMetalCommandQueue() { return ((UnityDisplaySurfaceMTL*)GetMainDisplaySurface())->commandQueue; }
 
-extern "C" EAGLContext*         UnityGetDataContextEAGL()   { return _GlesContext; }
+extern "C" EAGLContext*         UnityGetDataContextEAGL()   {
+    return _GlesContext;
+}
```

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60011-UNI-11095.mm" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60011-UNI-11095.mm"
index 4f88df5..33712e5 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60011-UNI-11095.mm"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60011-UNI-11095.mm"
@@ -1,15 +1,23 @@
 // Create oneliners
 
-extern "C" NSBundle*            UnityGetMetalBundle()       { return _MetalBundle; }
+extern "C" NSBundle*            UnityGetMetalBundle()       {
+    return _MetalBundle;
+}
 extern "C" MTLDeviceRef         UnityGetMetalDevice()       { return _MetalDevice; }
 extern "C" MTLCommandQueueRef   UnityGetMetalCommandQueue() { return ((UnityDisplaySurfaceMTL*)GetMainDisplaySurface())->commandQueue; }
 
-extern "C" EAGLContext*         UnityGetDataContextEAGL()   { return _GlesContext; }
+extern "C" EAGLContext*         UnityGetDataContextEAGL()   {
+    return _GlesContext;
+}
 
 // Keep oneliners
 
-extern "C" NSBundle*            UnityGetMetalBundle()       { return _MetalBundle; }
+extern "C" NSBundle*            UnityGetMetalBundle()       {
+    return _MetalBundle;
+}
 extern "C" MTLDeviceRef         UnityGetMetalDevice()       { return _MetalDevice; }
 extern "C" MTLCommandQueueRef   UnityGetMetalCommandQueue() { return ((UnityDisplaySurfaceMTL*)GetMainDisplaySurface())->commandQueue; }
 
-extern "C" EAGLContext*         UnityGetDataContextEAGL()   { return _GlesContext; }
+extern "C" EAGLContext*         UnityGetDataContextEAGL()   {
+    return _GlesContext;
+}
```

Debug support files:
---
[60011-UNI-11095.mm](https://gist.githubusercontent.com/mihaipopescu/725114b97297675110a0a9008de8791a/raw/4f88df5136f44c07075e35f35260f181edb45257/60011-UNI-11095.mm)
[60011-UNI-11095.mm.log](https://gist.githubusercontent.com/mihaipopescu/725114b97297675110a0a9008de8791a/raw/95612d1a95b8d92dcf0e8fc3d0599af384b15a64/60011-UNI-11095.mm.log)
[60011-UNI-11095.mm.unc](https://gist.githubusercontent.com/mihaipopescu/725114b97297675110a0a9008de8791a/raw/c5843afeb892f257204019e6e8c2d54a560e5890/60011-UNI-11095.mm.unc)
[Uncrustify.Cpp.cfg](https://gist.githubusercontent.com/mihaipopescu/725114b97297675110a0a9008de8791a/raw/23d62d9dba5caf352fdc1793792e98eed4b81929/Uncrustify.Cpp.cfg)
[UNI-11095.mm](https://gist.githubusercontent.com/mihaipopescu/725114b97297675110a0a9008de8791a/raw/34ed40507a292466bccad0149f253378d6a59fdb/UNI-11095.mm)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by CDanU** [comment]
Currently there are no `nl_create_*_one_liner` options for functions.
Bug: The pointer star is falsely detected as CT_ARITH.

**Comment by gmaurel** [comment]
Fixed at a4347fc761f54ec1c31c9076805475175cdd6b00, on Feb 10 19:20:30 2018
